src/main/java/bio/terra/common/tracing/TracingConfig.java

```diff
@@ -1,151 +1,36 @@
 import org.springframework.context.annotation.Configuration;
-import org.springframework.core.env.ConfigurableEnvironment;
-import org.springframework.web.context.ServletContextAware;
+import org.springframework.context.annotation.Primary;
 import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
-/**
- * Spring Configuration for Terra common tracing setup.
- *
- * <ul>
- *   <li>{@link OcHttpServletFilter}, encloses HTTP requests to the server in a per-request
- *       OpenCensus trace.
- *   <li>{@link RequestAttributeInterceptor}, which adds additional per-request annotations for each
- *       service request.
- *   <li>{@link StackdriverTraceExporter} configuration, which causes traces sampled by the service
- *       to be exported to Stackdriver aka Google Cloud Tracing.
- *   <li>{@link CensusSpringAspect}, a Spring aspect to enable using the {@link
- *       io.opencensus.contrib.spring.aop.Traced} annotation on Spring bean methods to add sub
- *       spans.
- * </ul>
- *
- * <p>To enable, add this package to Spring's component scan, e.g.
- * {@code @ComponentScan(basePackages = "bio.terra.common.tracing")}
- */
+/** Spring Configuration for Terra common tracing setup. */
 @Configuration
-@EnableConfigurationProperties(value = TracingProperties.class)
-public class TracingConfig implements InitializingBean, WebMvcConfigurer, ServletContextAware {
-  private static final Logger logger = LoggerFactory.getLogger(TracingConfig.class);
-
-  private final ConfigurableEnvironment configurableEnvironment;
-  private final TracingProperties tracingProperties;
-  private ServletContext servletContext;
-
-  @Autowired
-  public TracingConfig(
-      ConfigurableEnvironment configurableEnvironment, TracingProperties tracingProperties) {
-    this.configurableEnvironment = configurableEnvironment;
-    this.tracingProperties = tracingProperties;
-  }
-
-  @Override
-  public void setServletContext(ServletContext servletContext) {
-    this.servletContext = servletContext;
-  }
-
-  /**
-   * A bean to register the {@link OcHttpServletFilter} to enclose server requests in OpenCensus
-   * span scopes.
-   */
+public class TracingConfig implements WebMvcConfigurer {
   @Bean
-  public FilterRegistrationBean<OcHttpServletFilter> tracingServletFilter() {
-    // Sam only supports B3 format headers, so prefer those everywhere to the default TraceContext
-    // format.
-    servletContext.setAttribute(
-        OcHttpServletFilter.OC_TRACE_PROPAGATOR, Tracing.getPropagationComponent().getB3Format());
-    FilterRegistrationBean<OcHttpServletFilter> registration =
-        new FilterRegistrationBean<>(new OcHttpServletFilter());
-    registration.setUrlPatterns(tracingProperties.getUrlPatterns());
-    return registration;
+  @Primary
+  @ConditionalOnProperty(
+      name = "terra.common.tracing.stackdriverExportEnabled",
```

Comment:
`stackdriverExportEnabled` was deleted in TracingProperties, why include this here?

Reply:
This reads it directly from config, no need to have it in TracingProperties too